taca/testing/create_uppmax_like_env.py

```diff
@@ -0,0 +1,379 @@
+""" Load and parse configuration file
+"""
+
+import glob
+import logging
+from taca.utils.config import CONFIG
+from taca.utils import config as conf
+import couchdb
+import os
+import datetime
+from dateutil.relativedelta import relativedelta
+import random
+import subprocess
+
+
+logger = logging.getLogger(__name__)
+
+def touch(file):
+    open(file, "w").close()
+
+def setupServer(conf):
+    url="http://{0}:{1}@{2}:{3}".format(conf['username'], conf['password'], conf['url'], conf['port'])
+    return couchdb.Server(url)
+
+
+def create_FC(incoming_dir, run_name, samplesheet):
+    # create something like 160217_ST-E00201_0063_AHJHNYCCXX
+    if os.path.exists(run_name):
+        # this FC exists, skip it
+        return
+    path_to_fc = os.path.join(incoming_dir, run_name)
+    if not os.path.exists(path_to_fc):
+        os.mkdir(path_to_fc)
+    touch(os.path.join(path_to_fc, "RTAComplete.txt"))
+    # create folder Demultiplexing
+    if not os.path.exists(os.path.join(path_to_fc, "Demultiplexing")):
+        os.mkdir(os.path.join(path_to_fc, "Demultiplexing"))
+    # create folder Demultiplexing/Reports
+    if not os.path.exists(os.path.join(path_to_fc, "Demultiplexing", "Reports")):
+        os.mkdir(os.path.join(path_to_fc, "Demultiplexing", "Reports"))
+    # create folder Demultiplexing/Stats
+    if not os.path.exists(os.path.join(path_to_fc, "Demultiplexing", "Stats")):
+        os.mkdir(os.path.join(path_to_fc, "Demultiplexing", "Stats"))
+
+    #memorise SampleSheet stats
+    header = []
+    for key in samplesheet[0]:
+        header.append(key)
+    counter = 1
+    current_lane = ""
+    for line in samplesheet:
+        project_name = ""
+        if "Project" not in line:
+            project_name = line["Sample_Project"]
+        else:
+            project_name = line["Project"]
+        lane = line["Lane"]
+        if current_lane == "":
+            current_lane = lane
+        elif current_lane != lane:
+            counter = 1
```

Comment:
yes, bcl2fastq puts a counter to each sample demux in a lane.... it is really strange and I do not understand the reason behind it.... but I try to simulate it